### ISSUE TITLE
Error message is truncated and update is pending when error in pulumi up.

### ISSUE THREAD

**Issue by TechWatching** [pulumi/pulumi#3937]
Often when I do a `pulumi up` and there is an error in my infrastructure , the error message is truncated and the process seems stuck. I have to do a `Ctr+C` to finish the porcess.

Moreover the update on pulumi.com seems to be pending for a while instead of being failed. However I have the complete error message and all the logs that were truncated in my terminal.
![image](https://user-images.githubusercontent.com/15186176/74836521-d9b79480-531f-11ea-8305-78f8b7568137.png)

**Comment by leezen** [comment]
Would you be willing to post the logs and messages to this issue with any sensitive information redacted?

**Comment by TechWatching** [comment]
Yes sure, where can I retreive the logs ? What would be needed exactly ?

**Comment by TechWatching** [comment]
@leezen I still have the issue. Could you tell me where I can find the logs that you need for that?

**Comment by Frassle** [comment]
Logs can be printed by adding `--logtostderr -v9` to your pulumi command.

**Comment by TechWatching** [comment]
Thanks. Here are the logs @leezen . The process does not end, so I had to Ctrl +C.
[logs_pulumi.txt](https://github.com/pulumi/pulumi/files/8257804/logs_pulumi.txt)

**Comment by mikhailshilkov** [comment]
@TechWatching As far as I can see, these logs only span 15 seconds and end with an error `'vaultName' does not match expression '^[a-z` - are you sure this is the result you described above?

**Comment by TechWatching** [comment]
That's not the same stack (I don't think I still have the one from the original description) but that's the same problem. Every time my stack has an error when being deployed, it truncates the logs and sometimes like this one stays sucked until I Ctrl + C the process.

**Comment by mikhailshilkov** [comment]
Do you have a repro program that I could run myself to see this? It sounds really odd.

**Comment by TechWatching** [comment]
Yes. You can find the branch [here](https://github.com/TechWatching/AzureFunctionSQLBindings/tree/issue-3937) (folder eng/infra for the IaC program). 
I have just re-executed, this time the process was not stuck but the error message was truncated like previously:
![issue3937](https://user-images.githubusercontent.com/15186176/159018798-44e692b1-df36-48cd-8c12-1b3bfc9ad247.png)
